### PR TITLE
Remove Out-Of-Process test site fixture and make deployer check physical 

### PR DESCRIPTION
**Pull request by jkotalik**
For https://github.com/aspnet/AspNetCore-Internal/issues/1998, https://github.com/aspnet/AspNetCore-Internal/issues/1999, https://github.com/aspnet/aspnetcore, etc. On Win7, we are seeing the IIS server start with an incorrect physicalPath. We trusted that the IIS ServerManager would switch applicationhost.configs correctly, but it seems like there is a race between modifying the redirection.config and starting the site.

I also removed the OutOfProcess test site fixture. It was only used for 2 tests and it made it increasingly difficult to debug issues.

### CODE REVIEW

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs [hunk 1]
```diff
@@ -5,41 +5,54 @@
...
+            Assert.Equal("gzip", response.Content.Headers.ContentEncoding.Single());
+            Assert.Equal(
+                new byte[] {
+                    0x1F, 0x8B, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    0x04, 0x0A, 0x63, 0x60, 0xA0, 0x3D, 0x00, 0x00,
+                    0xCA, 0xC6, 0x88, 0x99, 0x64, 0x00, 0x00, 0x00 },
+                await response.Content.ReadAsByteArrayAsync());
         }
 
+
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs]: nit: newline

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs [hunk 2]
```diff
@@ -5,41 +5,54 @@
...
+using Microsoft.AspNetCore.Server.IntegrationTesting;
 using Microsoft.AspNetCore.Testing.xunit;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
 {
-    [Collection(IISCompressionSiteCollection.Name)]
-    public abstract class CompressionTests : FixtureLoggedTest
+    [Collection(PublishedSitesCollection.Name)]
+    public class CompressionOutOfProcessTests : IISFunctionalTestBase
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs]: Shouldn't the name stay `CompressionTests`

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/ServerAbortTests.cs [hunk 3]
```diff
@@ -4,51 +4,52 @@
...
-            _fixture = fixture;
+                var deploymentResult = await DeployAsync(deploymentParameters);
+
+                var response = await deploymentResult.HttpClient.GetAsync("/Abort").DefaultTimeout();
+
+                Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
+                // 0x80072f78 ERROR_HTTP_INVALID_SERVER_RESPONSE The server returned an invalid or unrecognized response
+                Assert.Contains("0x80072f78", await response.Content.ReadAsStringAsync());
+            }
+            catch (HttpRequestException)
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/ServerAbortTests.cs]: Is this catch needed?

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Utilities/IISTestSiteCollection.cs [hunk 4]
```diff
@@ -13,17 +13,4 @@ public class IISTestSiteCollection : ICollectionFixture<IISTestSiteFixture>
...
     }
-
-    [CollectionDefinition(Name)]
-    public class OutOfProcessTestSiteCollection : ICollectionFixture<OutOfProcessTestSiteFixture>
-    {
-        public const string Name = nameof(OutOfProcessTestSiteCollection);
-    }
-
-    [CollectionDefinition(Name)]
-    public class OutOfProcessV1TestSiteCollection : ICollectionFixture<OutOfProcessV1TestSiteFixture>
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Utilities/IISTestSiteCollection.cs]: Are we losing test coverage by removing this?

Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/Utilities/IISTestSiteCollection.cs]: Nope. OutOfProcV1 was removed from 3.0.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/ServerAbortTests.cs (hunk 3 above)
Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/ServerAbortTests.cs]: Yeah. OutOfProc can either fail from an exception being throw or a bad gateway.

**Comment by jkotalik** [comment]
I also think this will fix https://github.com/aspnet/AspNetCore-Internal/issues/1957.

File: src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs [hunk 5]
```diff
@@ -241,6 +241,11 @@ private void WaitUntilSiteStarted()
                 var site = serverManager.Sites.Single();
                 var appPool = serverManager.ApplicationPools.Single();
 
+                if (site.Applications.Single().VirtualDirectories.Single().PhysicalPath != contentRoot)
+                {
+                    throw new InvalidOperationException("Wrong physical path");
```
Comment (Tratcher) [inline src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs]: Include the expected and actual in the error.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**